### PR TITLE
feat: Implement usage statistics persistence and memory optimizations

### PR DESCRIPTION
**Pull request by crazyrabbit0**
## 📝 Summary
This PR introduces persistent storage for the usage statistics to ensure data is not lost across server restarts. Alongside the persistence feature, it includes critical memory and performance optimizations for the logger plugin to handle high-throughput scenarios without memory leaks or excessive CPU overhead.

### 💾 Usage Persistence
* Added a background persistence loop (`StartPersistence`) that periodically saves usage statistics snapshots to disk (defaulting to every 5 minutes and on graceful shutdown).
* Implemented snapshot restoration to seamlessly reload historical usage data when the server starts.
* Exposed `usage-dir` in `config.yaml` and as a `CLI_PROXY_USAGE_DIR` environment variable (with Docker Compose support) to configure where the `usage.json` data is saved.

### 🚀 Performance & Memory Optimizations
* **Memory Bounds:** Added a hard limit (max 1000 items) on stored `RequestDetail` entries per model to prevent unbounded memory growth in long-running processes.
* **Allocation Reductions:** Caches the `dayKey` (YYYY-MM-DD) generation to avoid the heavy allocation overhead of calling `time.Format` on every single request record.
* **Serialization Efficiency:** Rewrote the JSON serialization logic to stream directly from the read-locked state instead of performing a full deep-copy snapshot, significantly reducing allocation spikes during routine saves.

## ⚙️ Configuration
You can define the usage directory in `config.yaml`:
```yaml
# Usage statistics directory (supports ~ for home directory)
usage-dir: "./usage"
```
Or via Docker environment variables:
```yaml
environment:
  - CLI_PROXY_USAGE_DIR=/CLIProxyAPI/usage
volumes:
  - ./usage:/CLIProxyAPI/usage
```

### CODE REVIEW

File: internal/usage/logger_plugin.go [hunk 1]
```diff
@@ -221,6 +248,74 @@ func (s *RequestStatistics) updateAPIStats(stats *apiStats, model string, detail
 	modelStatsValue.TotalRequests++
 	modelStatsValue.TotalTokens += detail.Tokens.TotalTokens
 	modelStatsValue.Details = append(modelStatsValue.Details, detail)
+	const maxDetails = 1000
+	if len(modelStatsValue.Details) > maxDetails {
+		cutoff := len(modelStatsValue.Details) - maxDetails + (maxDetails / 10)
+		// Use copy to shift elements to the front, preserving the backing array's capacity
+		copy(modelStatsValue.Details, modelStatsValue.Details[cutoff:])
+		modelStatsValue.Details = modelStatsValue.Details[:len(modelStatsValue.Details)-cutoff]
+	}
```
[the remark this answers is not in the document]
Comment (crazyrabbit0) [reply internal/usage/logger_plugin.go:262]: done 👌

File: internal/usage/logger_plugin.go [hunk 2]
```diff
@@ -221,6 +248,74 @@ func (s *RequestStatistics) updateAPIStats(stats *apiStats, model string, detail
...
+	for k, v := range snapshot.TokensByDay {
+		s.tokensByDay[k] = v
+	}
+
+	s.tokensByHour = make(map[int]int64)
+	for k, v := range snapshot.TokensByHour {
+		var hour int
+		hour, _ = strconv.Atoi(k)
+		s.tokensByHour[hour] = v
+	}
```
[the remark this answers is not in the document]
Comment (crazyrabbit0) [reply internal/usage/logger_plugin.go:328]: done 👌

**Comment by crazyrabbit0** [comment]
> Quick review pass:
> 
> * Main risk area here is auth/session state and stale credential handling.
> * I didn’t see targeted regression coverage in the diff; please add or point CI at a focused test for the changed path in .gitignore, config.example.yaml, docker-compose.yml (+6 more).
> * Before merge, I’d smoke-test the behavior touched by .gitignore, config.example.yaml, docker-compose.yml (+6 more) with malformed input / retry / rollback cases, since that’s where this class of change usually breaks.

I've added internal/usage/persistence_test.go with targeted regression tests covering the edge cases you mentioned:
- malformed JSON inputs
- missing files
- unwritable directories (simulating rollback/failure paths)

The .gitignore and config files are already updated to handle the new persistent volume.

**Comment by luispater** [comment]
We have no plans to support this.

**Comment by crazyrabbit0** [comment]
Oh crap, that hurt, ok I guess 🥲
